src/parsers/psse_dynamic_data.jl

```diff
@@ -0,0 +1,267 @@
+"""
+Parse .dyr file into a dictionary indexed by bus number.
+Each bus number key has a dictionary indexed by component type and id.
+
+"""
+function _parse_dyr_file(file::AbstractString)
+    dyr_text = read(file, String)
+    start = 1
+    parsed_values = Dict{Int, Dict}()
+    models = Vector{String}()
+    while start < length(dyr_text)
+        val = findnext('/', dyr_text, start)
+        if isnothing(val)
+            break
+        end
+        line = replace(dyr_text[start:(val - 1)], "\'" => "")
+        val_array = strip.(split(line))
```

Comment:
It would be good to `@assert` on the required size of this array and print the actual if it does trip.

Reply:
In `.dyr` files this is not possible to check. Depending on the component the size of the array is different (for example in GENCLS the length will be 5, and in GENROU will be 18). However, the first three strings are the same (bus, model and id).